src/Lean/Elab/MutualDef.lean

```diff
@@ -399,18 +404,19 @@ private def elabFunValues (headers : Array DefViewElabHeader) (vars : Array Expr
         -- leads to more section variables being included than necessary
         let val ← instantiateMVarsProfiling val
         let val ← mkLambdaFVars xs val
-        unless header.type.hasSorry || val.hasSorry do
+        if linter.unusedSectionVars.get (← getOptions) && !header.type.hasSorry && !val.hasSorry then
           for var in vars do
             unless header.type.containsFVar var.fvarId! ||
                 val.containsFVar var.fvarId! ||
                 (← vars.anyM (fun v => return (← v.fvarId!.getType).containsFVar var.fvarId!)) do
               let varDecl ← var.fvarId!.getDecl
               let var := if varDecl.userName.hasMacroScopes && varDecl.binderInfo.isInstImplicit then
-                m!"[{varDecl.type}]".group
+                m!"{indentD m!"[{varDecl.type}]"}{Format.line}".group
               else
-                var
-              logWarningAt header.ref m!"included section variable '{var}' is not used in \
-                '{header.declName}', consider excluding it"
+                m!" '{var}' "
+              Linter.logLint linter.unusedSectionVars header.ref
+                m!"included section variable{var}is unused in theorem, consider restructuring \
```

Comment:
Is the change from `header.declName` to `theorem` intentional? At least the error message itself being much less useful. (Or is the reasoning that the linter underlining a particular place will make up for this? In this case, I'd at least write "this theorem".)

Reply:
Yes, the warning position is the indicator as properly referring to the theorem name at this point is not possible

Reply:
I see. Can you write "this theorem" then?
```suggestion
                m!"included section variable{var}is unused in this theorem, consider restructuring \